### PR TITLE
Video code enhancements

### PR DESCRIPTION
**Pull request by mohamedchebbii**
- Rework of aspectRatio method for all video formats
- Rework of GUID's asf decoding

### CODE REVIEW

**Comment by neheb** [comment]
```
 D:\a\exiv2\exiv2\src\helper_functions.cpp(68): error C2220: the following warning is treated as an error
D:\a\exiv2\exiv2\src\helper_functions.cpp(68): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data
D:\a\exiv2\exiv2\src\helper_functions.cpp(69): warning C4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data
```

File: src/asfvideo.cpp [hunk 1]
```diff
@@ -21,149 +25,177 @@ namespace Exiv2::Internal {
...
+ * identify an object across different systems. The GUID structure is comprised of four 32-bit values and an
+ * array of 8 bytes.
+ *
+ * @note The byte order of the GUID structure is in little endian.
+ *
+ * @see https://en.wikipedia.org/wiki/Globally_unique_identifier
+ *
+ */
+
+bool AsfVideo::GUIDTag::operator==(const AsfVideo::GUIDTag& other) const {
```
Comment (neheb) [inline src/asfvideo.cpp:44]: This is equivalent to = default. No need for this.

File: include/exiv2/asfvideo.hpp [hunk 2]
```diff
@@ -47,6 +48,41 @@ class EXIV2API AsfVideo : public Image {
...
+    uint32_t data1_;
+    uint16_t data2_;
+    uint16_t data3_;
+    std::array<byte, 8> data4_;
+
+   public:
+    bool operator==(const GUIDTag& other) const;
+
+    // Constructor to create a GUID object by passing individual values for each attribute
+    GUIDTag(unsigned int data1, unsigned short data2, unsigned short data3, std::array<byte, 8> data4);
```
Comment (neheb) [inline include/exiv2/asfvideo.hpp:77]: I don't think this is needed, {} can be used instead of ().

Comment (mohamedchebbii) [reply include/exiv2/asfvideo.hpp:77]: but here I already have " GUIDTag(const uint8_t* bytes);" witch will delete the default initializer list

```
exiv2/src/asfvideo.cpp:92:25: error: no matching constructor for initialization of 'const AsfVideo::GUIDTag'
const AsfVideo::GUIDTag Header = {0x75B22630, 0x668E, 0x11CF, {0xA6, 0xD9, 0x00, 0xAA, 0x00, 0x62, 0xCE, 0x6C}};
```

File: src/asfvideo.cpp (hunk 1 above)
Comment (mohamedchebbii) [reply src/asfvideo.cpp:44]: I beleive you are talking about operator= but here I developped operator== to compare with other tags.
```
github/exiv2/src/asfvideo.cpp:203:17: error: invalid operands to binary expression ('const AsfVideo::GUIDTag' and 'AsfVideo::GUIDTag')
  return Header == AsfVideo::GUIDTag(buf);
```

File: include/exiv2/asfvideo.hpp (hunk 2 above)
Comment (neheb) [reply include/exiv2/asfvideo.hpp:77]: that error comes from the std::array.

```
const AsfVideo::GUIDTag Header = {0x75B22630, 0x668E, 0x11CF, std::array<byte, 8>{0xA6, 0xD9, 0x00, 0xAA, 0x00, 0x62, 0xCE, 0x6C}};
```

should work. or don't use std::array. I can handle this in a later PR if you want.

**Review by neheb — APPROVED** [review]

**Pull request merged**